internal/serve/serve.go

```diff
@@ -63,10 +63,21 @@ func getHandlerDeps(cfg Configs) (handlerDeps, error) {
 
 func handler(deps handlerDeps) http.Handler {
 	mux := supporthttp.NewAPIMux(deps.Logger)
-	mux.NotFound(errorHandler{Error: notFound}.ServeHTTP)
-	mux.MethodNotAllowed(errorHandler{Error: methodNotAllowed}.ServeHTTP)
+	mux.NotFound(httperror.ErrorHandler{Error: httperror.NotFound}.ServeHTTP)
+	mux.MethodNotAllowed(httperror.ErrorHandler{Error: httperror.MethodNotAllowed}.ServeHTTP)
 
 	mux.Get("/health", health.PassHandler{}.ServeHTTP)
 
+	// Authenticated routes
+	mux.Group(func(r chi.Router) {
+		// r.Use(...authMiddleware...)
+
+		r.Route("/payments", func(r chi.Router) {
+			handler := &httphandler.PaymentsHandler{}
```

Comment:
Do we not have to put models in the handler?

Reply:
We do indeed, good catch! 🚀